actions/image/crop.php

```diff
@@ -0,0 +1,82 @@
+<?php
+/**
+ * Image crop action
+ */
+
+$guid = get_input('guid');
+$entity = get_entity($guid);
+
+if (!$entity || !($entity instanceof ElggEntity) || !$entity->canEdit()) {
+	register_error(elgg_echo('image:crop:fail'));
+	forward(REFERER);
+}
+
+$x1 = (int) get_input('x1', 0);
+$y1 = (int) get_input('y1', 0);
+$x2 = (int) get_input('x2', 0);
+$y2 = (int) get_input('y2', 0);
+
+$filehandler = new ElggFile();
+$filehandler->owner_guid = $entity->getGUID();
+$filehandler->setFilename("icon/master.jpg");
```

Comment:
So your comment was positive "I wish all entities were doing it like this"?

I'm aware of the madness. Once we agree that ```<guid>/icon/<size>.jpg``` is The right way™ we can update all entities to use it.